.github/workflows/docker-build-push.yml

```diff
@@ -12,10 +12,20 @@ on:
         required: false
         type: string
         default: ./docker/Dockerfile
+    secrets:
+      DOCKERHUB_USERNAME:
+        description: The Docker Hub username.
+        required: true
+      DOCKERHUB_TOKEN:
+        description: A token to push images to the Docker Hub repository.
+        required: true
+      BUILD_ARGS:
+        description: Variables which will be passed as build arguments to the Dockerfile.
+        required: false
 
 jobs:
   production-image:
-    if: ${{ github.event_name == 'release' }}
+    if: ${{ github.event_name == 'release' || (github.event_name == 'push' && github.ref_name == 'release') }}
```

Comment:
whats the background for always creating an image if something gets committed into release?